### ISSUE TITLE
RFE: Add Accessibility features

### ISSUE THREAD

**Issue by ghost** [codenameone/CodenameOne#426]
Original [issue 426](https://code.google.com/p/codenameone/issues/detail?id=426) created by codenameone on 2012-12-11T19:21:49.000Z:

Currently there's no mechanism to support visually impaired users.  If, for example, an Android user has TalkBack enabled (Settings, Accessibility, TalkBack -&gt; enable), no audible cues or haptic feedback occurs in a Codename One app.

Related links:

http://developer.android.com/guide/topics/ui/accessibility/apps.html

**Comment by codenameone** [comment]
This is something that was requested by some vendors as it's essential in some locales for government and anti-discrimination.

**Comment by codenameone** [comment]
I've started working on this but had to scrap it due to lack of time. This should be relatively simple at least on Android with this guide https://developer.android.com/guide/topics/ui/accessibility/custom-views.html

We already support a lot of the core features such as the ability to navigate with keys instead of touch.

**Comment by codenameone** [comment]
I wanted to get this through but it's hard to allocate time for this right now.

**Comment by flewby** [comment]
I would say this is a pretty important feature to add, for the reasons already given. Looking forward to being able to have an accessible app!

**Comment by codenameone** [comment]
We agree. Unfortunately this is a non-trivial issue to say the least...

**Comment by kutoman** [comment]
yeah we also are getting this request. Due to this lacking it might happen that we have to switch to another cross-platform solution in the future unfortunately.

**Comment by ionlake** [comment]
we have committed to some companies demanding this that we will have these features by mid 2021. Based on this thread, it appears this may not happen and just like a previous message, we will be forced to find another vendor. If you cannot allocate time to get this done, i would love to know now as a complete change over will take time and I don't want to have to do this at the last minute.

**Comment by shai-almog** [comment]
This is scheduled for mid 2021 but it's hard for us to make guarantees on such a complex issue. Currently the plan is to have Android support by then. I'm not sure when the iOS support will land.

**Comment by flewby** [comment]
Any progress on adding accessibility to Android?

**Comment by shai-almog** [comment]
We have a person "working on it" but I'm not too optimistic as I haven't seen any code yet so I'm not sure where this stands.

**Comment by CoderX-man** [comment]
hello!!
I'm a newbie. I want to work on this project. Can I start working on it or is this project already assigned to someone else?

**Comment by shai-almog** [comment]
Hi,
It isn't assigned to anyone right now.

**Comment by CoderX-man** [comment]
Can I start working on it?

**Comment by shai-almog** [comment]
Sure.

**Comment by pranavbalakrishnan4100** [comment]
Hi @shai-almog. If there hasn't been any progress reg this so far, please let me know if it's possible to me assign this to me. I would like to work on it.

**Comment by shai-almog** [comment]
Hi @pranavbalakrishnan4100 
Sure. Assigned you.

**Comment by sri0781** [comment]
@pranavbalakrishnan4100   I would like to know any progress made on enabling accessibility to android?
